### PR TITLE
[WebApp] Add passkey support logic

### PR DESCRIPTION
**Pull request by Maks19**
**Description:** Add passkey support logic

**Issue:** https://www.notion.so/saladtech/FE-Web-app-Passkey-support-logic-ecedbebec44d43eebdbb883a9ed08c01

### CODE REVIEW

File: packages/web-app/src/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    } catch (error) {
+      console.error('Error checking passkey support:', error)
+      return false
+    }
+  }
+
+  return false
+}
+
+export const usePasskeySupport = (): boolean => {
```
Comment (vitto-moz) [inline packages/web-app/src/hooks.ts:28]:
```suggestion
export const useWithPasskeySupport = (): boolean => {
```

File: packages/web-app/src/hooks.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from 'react'
+
+const isDeviceSupportPasskey = async (): Promise<boolean> => {
```
Comment (vitto-moz) [inline packages/web-app/src/hooks.ts:3]:
```suggestion
const getWithPasskeySupport= async (): Promise<boolean> => {
```

File: packages/web-app/src/hooks.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      console.error('Error checking passkey support:', error)
+      return false
+    }
+  }
+
+  return false
+}
+
+export const usePasskeySupport = (): boolean => {
+  const [isPasskeySupported, setIsPasskeySupported] = useState<boolean>(false);
```
Comment (vitto-moz) [inline packages/web-app/src/hooks.ts:29]:
```suggestion
  const [withPasskeySupport, setWithPasskeySupport] = useState<boolean>(false);
```

**Review by vitto-moz — CHANGES_REQUESTED** [review]

**Comment by vitto-moz** [comment]
I'm not sure about idea to cover this functionality with hook, as the passkey support doesn't change during user session, does it? 
Seems it's enough to make the check once in the related Store part, during initialization and then just use this value wherever we need it across the app.

**Comment by Maks19** [comment]
> I'm not sure about idea to cover this functionality with hook, as the passkey support doesn't change during user session, does it? Seems it's enough to make the check once in the related Store part, during initialization and then just use this value wherever we need it across the app.

Changed it to Store

File: packages/web-app/src/modules/passkey-setup/utils.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+export const getIsDeviceSupportPasskey = async (): Promise<boolean> => {
```
Comment (vitto-moz) [inline packages/web-app/src/modules/passkey-setup/utils.ts:1]:
```suggestion
export const getPasskeySupported= async (): Promise<boolean> => {
```
or 

```suggestion
export const getWithPasskeySupport= async (): Promise<boolean> => {
```

File: packages/web-app/src/modules/passkey-setup/PasskeyStore.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { action, observable, runInAction } from 'mobx'
+import { getIsPasskeySupported } from './utils'
+
+export class PasskeyStore {
+  @observable
+  public isPasskeySupported: boolean = false
+
+  constructor() {
+    this.setIsPasskeySupport()
```
Comment (vitto-moz) [inline packages/web-app/src/modules/passkey-setup/PasskeyStore.tsx:9]:
```suggestion
    this.setIsPasskeySupported()
```

File: packages/web-app/src/modules/passkey-setup/PasskeyStore.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  }
+
+  @action
+  private async setIsPasskeySupport(): Promise<void> {
+    try {
+      const isSupported = await getIsPasskeySupported()
```
Comment (vitto-moz) [inline packages/web-app/src/modules/passkey-setup/PasskeyStore.tsx:15]:
```suggestion
      const isPasskeySupported= await getIsPasskeySupported()
```

File: packages/web-app/src/modules/passkey-setup/PasskeyStore.tsx (hunk 5 above)
Comment (Maks19) [reply packages/web-app/src/modules/passkey-setup/PasskeyStore.tsx]: Fixed

File: packages/web-app/src/modules/passkey-setup/PasskeyStore.tsx (hunk 6 above)
Comment (Maks19) [reply packages/web-app/src/modules/passkey-setup/PasskeyStore.tsx:15]: Fixed

**Review by vitto-moz — APPROVED** [review]

**Pull request merged**